dts/bindings/video/video-interfaces.yaml

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for video interfaces.
+
+include: base.yaml
+
+properties:
+  bus-type:
+    type: int
+    enum:
+      - 1 # Parallel
+      - 2 # MIPI CSI-2 D-PHY
+      - 3 # MIPI CSI-2 C-PHY
+      - 4 # MIPI CSI-1
+    description:
+      Data bus type.
+
+  bus-width:
+    type: int
+    maximum: 64
+    description:
+      Number of data lines actively used, valid for the parallel busses.
+
+  data-shift:
+    type: int
+    maximum: 64
+    description:
+      On the parallel data busses, if bus-width is used to specify the number of
+      data lines, data-shift can be used to specify which data lines are used,
+      e.g. "bus-width=<8>; data-shift=<2>;" means, that lines 9:2 are used.
```

Comment:
I like `data-lanes` as it is less ambiguous and helps with vendors choosing 0- or 1-indexing. Would you know why parallel port do not use them in Linux? I do not find anything outside the original commit that introduced it.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/Documentation/devicetree/bindings/media/video-interfaces.txt?h=53c5b6c95f1850b0f95b445e0c6f1c133b00bfef
On the other hand, keeping as much compatibility with Linux syntax also helps with portability.